app/modules/tmr.c

```diff
@@ -307,8 +340,44 @@ static int tmr_softwd( lua_State* L ){
 	return 0; 
 }
 
+// Lua: tmr.create()
+static int tmr_create( lua_State *L ) {
+  timer_t ud = (timer_t)lua_newuserdata(L, sizeof(timer_struct_t));
+  if (!ud) return luaL_error(L, "not enough memory");
+  luaL_getmetatable(L, "tmr.timer");
+  lua_setmetatable(L, -2);
+  ud->lua_ref = LUA_NOREF;
+  ud->self_ref = LUA_NOREF;
+  ud->mode = TIMER_MODE_OFF;
+  ets_timer_disarm(&ud->os);
+  return 1;
+}
+
+// Lua: tmr.delete()
+static int tmr_delete( lua_State *L ) {
+  timer_t ud = (timer_t)luaL_checkudata(L, 1, "tmr.timer");
+  if (!ud) return luaL_error(L, "wrong userdata");
+  ets_timer_disarm(&ud->os);
+  luaL_unref(L, LUA_REGISTRYINDEX, ud->self_ref); ud->self_ref = LUA_NOREF;
+  luaL_unref(L, LUA_REGISTRYINDEX, ud->lua_ref); ud->lua_ref = LUA_NOREF;
+  return 0;
+}
+
 // Module function map
 
+static const LUA_REG_TYPE tmr_dyn_map[] = {
+	{ LSTRKEY( "register" ),     LFUNCVAL( tmr_register ) },
+	{ LSTRKEY( "alarm" ),        LFUNCVAL( tmr_alarm ) },
+	{ LSTRKEY( "start" ),        LFUNCVAL( tmr_start ) },
+	{ LSTRKEY( "stop" ),         LFUNCVAL( tmr_stop ) },
+	{ LSTRKEY( "unregister" ),   LFUNCVAL( tmr_unregister ) },
+	{ LSTRKEY( "state" ),        LFUNCVAL( tmr_state ) },
+	{ LSTRKEY( "interval" ),     LFUNCVAL( tmr_interval) }, 
+  { LSTRKEY( "__gc" ),         LFUNCVAL( tmr_delete ) },
```

Comment:
Why not just call `tmr_unregister()` upon garbage collection? IMO the only difference is the condition for `ets_timer_disarm()`.